### PR TITLE
Honor ac_exclude / ac_include in autodiscovery

### PR DESCRIPTION
**Pull request by xvello**
### What does this PR do?

Make the three AD listeners honor the container filter settings

### Motivation

Causes a regression in logs container discovery

### Testing

Image `datadog/agent-dev:xvello-ac-exclude` tested on vanilla docker, fargate and gke

### CODE REVIEW

File: pkg/autodiscovery/listeners/ecs.go [hunk 1]
```diff
@@ -116,6 +122,10 @@ func (l *ECSListener) refreshServices(firstRun bool) {
 			log.Debugf("container %s is in status %s - skipping", c.DockerID, c.KnownStatus)
 			continue
 		}
+		if l.filter.IsExcluded(c.DockerName, c.Image) {
+			log.Debugf("container %s is filtered out", c.DockerID)
```
Comment (mfpierre) [inline pkg/autodiscovery/listeners/ecs.go]: ditto

File: pkg/autodiscovery/listeners/kubelet.go [hunk 2]
```diff
@@ -139,6 +145,10 @@ func (l *KubeletListener) createService(entity string, pod *kubelet.Pod, firstRu
 	var containerName string
 	for _, container := range pod.Status.Containers {
 		if container.ID == svc.entity {
+			if l.filter.IsExcluded(container.Name, container.Image) {
+				log.Debugf("container %s is filtered out", container.ID)
```
Comment (mfpierre) [inline pkg/autodiscovery/listeners/kubelet.go]: ditto

**Review by mfpierre — DISMISSED** [review]
LGTM small nits on troubleshooting logs

File: pkg/autodiscovery/listeners/docker.go [hunk 3]
```diff
@@ -184,8 +195,19 @@ func (l *DockerListener) createService(cID string) {
...
-	} else if findKubernetesInLabels(cInspect.Config.Labels) {
-		isKube = true
+	} else {
+		image, err := l.dockerUtil.ResolveImageName(cInspect.Image)
+		if err != nil {
+			log.Warnf("error while resolving image name: %s", err)
+			image = ""
+		}
+		if l.filter.IsExcluded(cInspect.Name, image) {
+			log.Debugf("container %s is filtered out", cInspect.ID)
```
[the remark this answers is not in the document]
Comment (xvello) [reply pkg/autodiscovery/listeners/docker.go]: Changed, output is now:
> 2018-09-10 12:56:12 UTC | DEBUG | (docker.go:337 in isExcluded) | container ef6627827900 filtered out: name "/includ" image "redis"

[the remark this answers is not in the document]
Comment (xvello) [reply pkg/autodiscovery/listeners/docker.go]: This specific case will not log anything though, as our dockerutils already filter out events from excluded events

**Review by hkaj — DISMISSED** [review]
👍

**Comment by hkaj** [comment]
unit test failure seems unrelated, integration test might be legit though:
```
--- FAIL: TestDockerListenerSuite (11.43s)
    --- FAIL: TestDockerListenerSuite/TestListenAfterStart (5.07s)
    	testing.go:730: race detected during execution of test
	testing.go:730: race detected during execution of test```

**Review by mfpierre — APPROVED** [review]
:shipit:

**Pull request merged**